src/core/lv_obj_pos.c

```diff
@@ -154,6 +138,19 @@ bool lv_obj_refr_size(lv_obj_t * obj)
         int32_t minw = calc_dynamic_width(obj, lv_obj_get_style_min_width(obj, LV_PART_MAIN), &content_width);
         int32_t maxw = calc_dynamic_width(obj, lv_obj_get_style_max_width(obj, LV_PART_MAIN), &content_width);
         w = LV_CLAMP(minw, w, maxw);
+
+        /**
+         * If the object style (after clamping) results in a width that is defined as a percentage of the parent,
+         * and if the parent's width is set to LV_SIZE_CONTENT and not managed by a layout, this object should not
+         * influence the parent's content width calculation. Thus, the `h_ignore_size` flag is set accordingly.
+         */
+        int32_t w_style = w == minw ? lv_obj_get_style_min_width(obj, LV_PART_MAIN)
+                          : (w == maxw ? lv_obj_get_style_max_width(obj, LV_PART_MAIN)
+                             : lv_obj_get_style_width(obj, LV_PART_MAIN));
+        obj->w_ignore_size =
+            (LV_COORD_IS_PCT(w_style) && parent->w_layout == 0 && lv_obj_get_style_width(parent, 0) == LV_SIZE_CONTENT)
+            ? 1
+            : 0;
```

Comment:
Ternary condition can be removed

```suggestion
        obj->w_ignore_size =
            LV_COORD_IS_PCT(w_style) && parent->w_layout == 0 && lv_obj_get_style_width(parent, 0) == LV_SIZE_CONTENT;
```